packages/adminrouter/extra/src/README.md

```diff
@@ -264,28 +264,178 @@ it, but it does not seem to be justified. For example:
+  and these lebels:
+  * `DCOS_SERVICE_NAME` label contents is equal to given `namespace+service` string
+  * `DCOS_SERVICE_SCHEME` label is set
+  * `DCOS_SERVICE_PORT_INDEX` label is set
+  * task is in state `TASK_RUNNING`
+  If all the conditions above are met, request is routed using root Marathon
+  data. Please check the [section below](#recommended-way-to-expose-services)
+  for details on the meaning of these labels.
+* if the name of the service couldn't be resolved using root Marathon data, AR
+  scans the list of Mesos frameworks in search for the one with either framework
+  ID (higher priority) or framework name equal(lower priority) to
+  `namespace+service` string. If found, depending on the value of `webui_url` field:
+  * field contains an empty string - algorithm tries to resolve the
+    `namespace+service` name using MesosDNS
+  * request is proxied to the address pointed by the field contents
+  * field is `null` - resolving process for given `namespace and name` ends, MesosDNS
+    step is NOT executed
+* resolving via MesosDNS relies on issuing an HTTP SRV query to MesosDNS. For
+  `namespace and name` string equal to `foo/bar/baz`, the request that is going to
+  be made is:
+  ```
+  http://localhost:8123/baz.bar.foo._tcp.marathon.mesos
+  ```
+  In case when MesosDNS returns no results, AR assumes that service
+  `namespace and name` does run on DC/OS.
+
+#### Stale cache/broken cache cases
+`/service` endpoint internally uses AR cache, so an AR failing to refresh it
+has the same effects on it like any other endpoint. Please check the [Cache
+timers](#cache-timers) section for details.
+
+The thing worth remembering though, is that this endpoint uses more than one cache
+entry. Because of that, if e.g. root Marathon fails, and Mesos is fine, requests
+will not be routed. This is because there has to be a strict priority enforcement,
+and without root Marathon, it is impossible to decide whether AR should use Mesos
+data or not (-> request routing ambiguity). The reverse is possible though -
+failed Mesos cache update will not prevent requests from being routed as long
+as they can be handled *only* using Marathon data.
+
+### Limitations
+Apart from the one mentioned earlier - that `/service` endpoint should be routing
+only to the tasks launched by root Marathon, there is one more limitation that
+needs to be kept in mind.
+
+To properly remove framework, a cleanup procedure needs to be followed:
+* framework needs to be removed from root Marathon
+* it needs to be unregistered with Mesos
+* Zookeeper state has to be cleaned up
+* Mesos reservations (if any), need to be deleted
+
+Without it Mesos framework is still present in the output of `/state-summary`
+endpoint and thus Admin Router will try to route it instead of replying with 404.
+It is impossible for AR to differentiate between frameworks that:
+* have been "just" removed/deleted by Marathon
```

Comment:
Does "just" mean "only" or "recently"? I think use one of those to avoid ambiguity - I think "recently"